presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -313,6 +315,34 @@ protected Boolean visitInPredicate(InPredicate node, Void context)
         protected Boolean visitFunctionCall(FunctionCall node, Void context)
         {
             if (metadata.isAggregationFunction(node.getName())) {
+                if (node.getOrderBy().isPresent()) {
+                    List<Expression> sortKeys = node.getOrderBy().get().getSortItems().stream()
+                            .map(SortItem::getSortKey)
+                            .collect(toImmutableList());
+                    if (node.isDistinct()) {
+                        List<FieldId> argumentReferences = node.getArguments().stream()
+                                .map(NodeRef::of)
+                                .map(columnReferences::get)
+                                .collect(toImmutableList());
+                        for (Expression sortKey : sortKeys) {
+                            if (!argumentReferences.contains(columnReferences.get(NodeRef.of(sortKey)))) {
+                                throw new SemanticException(
+                                        ORDER_BY_MUST_BE_IN_AGGREGATE,
+                                        sortKey,
+                                        "For aggregate function with DISTINCT, ORDER BY expressions must appear in arguments");
+                            }
+                        }
+                    }
+                    // ensure that no output fields are referenced from ORDER BY clause
```

Comment:
Could you please help me with an example that aggregation would refer a output column and won't be caught by something like "column not found". I sort of feel that this might still be necessary but don't really know why...